src/util/mod.rs

```diff
@@ -683,6 +683,114 @@ pub(crate) unsafe fn copy_unchecked(src: &[u8], dst: &mut [u8]) {
     };
 }
 
+/// Unsafely transmutes the given `src` into a type `Dst`.
+///
+/// # Safety
+///
+/// The value `src` must be a bit-valid instance of `Dst`.
+#[inline(always)]
+pub(crate) const unsafe fn transmute_unchecked<Src, Dst>(src: Src) -> Dst {
+    static_assert!(Src, Dst => core::mem::size_of::<Src>() == core::mem::size_of::<Dst>());
```

Comment:
Mention this requirement in the doc comment?

Reply:
It's present as a debugging safeguard, but not something callers may rely upon. The onus is entirely on callers to prove that `src` is a bit-valid instance of `Dst`. Those safety proofs will remain correct even if we remove this `static_assert` which we will probably eventually do to support truncating and padding-extension transmutations.

Reply:
Ack